addons/pkg/util/cluster_variable_util.go

```diff
@@ -29,6 +30,41 @@ func ParseClusterVariableString(cluster *clusterapiv1beta1.Cluster, variableName
 	return result.(string), err
 }
 
+func ParseClusterVariableInterface(cluster *clusterapiv1beta1.Cluster, variableName, keyName string) (string, error) {
+	var result interface{}
+
+	result, err := parseClusterVariable(cluster, variableName)
+	if err != nil || result == nil {
+		return "", err
+	}
+	interfaceVars := result.(map[string]interface{})
+
+	if valueName, ok := interfaceVars[keyName]; ok {
+		return valueName.(string), err
```

Comment:
Can we add a type check before return `valueName.(string)`? Possibly valueName might not be a string?

Reply:
added type check before return